api/app/src/shared/analytics.ts

```diff
@@ -0,0 +1,32 @@
+import { PostHog } from "posthog-node";
+import { Request } from "express";
+
+import { Config } from "./config";
+import { getCxId } from "../routes/util";
+
+export const queryAnalytics = ({
+  message,
+  req,
+  apiType,
+}: {
+  message: string;
```

Comment:
If I understand it correctly, this is the type of event on posthog, right?

In theory, we would like to have a finite universe of events, so maybe we add a strong type, maybe with a string literal union?